signac/core/synced_collection.py

```diff
@@ -106,28 +122,62 @@ def _load(self):
         pass
 
     @abstractmethod
-    def _sync(self):
+    def _sync(self, data):
         """Write data to underlying backend."""
         pass
 
+    @abstractmethod
+    def write_to_cache(self, data=None):
+        """Write data to cache."""
+        pass
+
+    @abstractmethod
+    def read_from_cache(self):
+        """Read data from cache."""
+        pass
+
+    def _sync_to_backend(self):
+        self._sync()
+
     def sync(self):
         """Synchronize the data with the underlying backend."""
         if self._suspend_sync_ <= 0:
             if self._parent is None:
-                self._sync()
+                self._sync_to_backend()
+                self.write_to_cache()
             else:
                 self._parent.sync()
 
     def load(self):
         """Load the data from the underlying backend."""
         if self._suspend_sync_ <= 0:
             if self._parent is None:
-                data = self._load()
+                data = self.read_from_cache()
```

Comment:
In my understanding, this order of operations means that the _cache is treated as the "primary" source of truth_. That is, (assuming non-buffered mode) if a file is modified outside of this process, I understand that the Redis cache takes precedence over the file's contents. Is that correct? (I think that's how the cache is designed to work, just wanted to verify my understanding.)

Do we need to add a way to [explicitly purge, refresh, or ban elements of the cache](https://en.wikipedia.org/wiki/Cache_invalidation)? (Only one of those methods purge/refresh/ban would be necessary, not all of them.)

Reply:
Yes, that is correct. 
>Do we need to add a way to explicitly purge, refresh, or ban elements of the cache? (Only one of those methods purge/refresh/ban would be necessary, not all of them.)

 I will add that.